dev/src/Command/AddComponentCommand.php

```diff
@@ -184,21 +210,215 @@ protected function execute(InputInterface $input, OutputInterface $output)
+    }
+
+    private function owlbotCopyCode(string $componentName, string $googleApisGenDir): string
+    {
+        list($userId, $groupId) = $this->getUserAndGroupId();
+        $command = [
+            'docker',
+            'run',
+            '--rm',
+            '--user',
+            sprintf('%s::%s', $userId, $groupId),
+            '-v',
+            $this->rootPath . ':/repo',
+            '-v',
+            $googleApisGenDir . ':/googleapis-gen',
+            '-w',
+            '/repo',
+            '--env',
+            'HOME=/tmp',
+            self::OWLBOT_CLI_IMAGE,
+            'copy-code',
+            sprintf('--config-file=%s/.OwlBot.yaml', $componentName),
+            '--source-repo=/googleapis-gen'
+        ];
+        return $this->runCommand($command);
+    }
+
+    private function owlbotCopyBazelBin(string $componentName, string $googleApisDir): string
+    {
+        list($userId, $groupId) = $this->getUserAndGroupId();
+        $command = [
+            'docker',
+            'run',
+            '--rm',
+            '--user',
+            sprintf('%s::%s', $userId, $groupId),
+            '-v',
+            $this->rootPath . ':/repo',
+            '-v',
+            $googleApisDir . '/bazel-bin:/bazel-bin',
+            self::OWLBOT_CLI_IMAGE,
+            'copy-bazel-bin',
+            sprintf('--config-file=%s/.OwlBot.yaml', $componentName),
+            '--source-dir',
+            '/bazel-bin',
+            '--dest',
+            '/repo'
+        ];
+        return $this->runCommand($command);
+    }
+
+    private function owlbotPostProcessor(): string
+    {
+        list($userId, $groupId) = $this->getUserAndGroupId();
+        $command = [
+            'docker',
+            'run',
+            '--rm',
+            '--user',
+            sprintf('%s::%s', $userId, $groupId),
+            '-v',
+            $this->rootPath . ':/repo',
+            '-w',
+            '/repo',
+            self::OWLBOT_PHP_IMAGE
+        ];
+        return $this->runCommand($command, null, null, true);
+    }
+
+    private function isProductHomePageMissing(string $productHomePage): bool
+    {
+        $response = $this->httpClient->request('GET', $productHomePage, ['http_errors' => false]);
+        return $response->getStatusCode() >= 400;
+    }
+
+    private function runCommand(
+        array $command,
+        ?string $workDir = null,
+        ?string $input = null,
+        bool $errorOutput = false
+    ): string {
+        $process = new Process($command);
+        if (!is_null($workDir)) {
+            $process->setWorkingDirectory($workDir);
+        }
+        if (!is_null($input)) {
+            $process->setInput($input);
+        }
+        // `mustRun` will throw a ProcessFailedException if the process
+        // couldn't be executed successfully.
+        $process->mustRun();
+        if ($errorOutput) {
+            return $process->getErrorOutput();
+        }
+        return $process->getOutput();
```

Comment:
I'm still a little confused as to what this does - if we want the error output, should we just concatenate it to the regular output?
```suggestion
        return $process->getOutput() . $process->getErrorOutput();
```